test/integration/targets/azure_rm_managed_disk/tasks/main.yml

```diff
@@ -190,6 +183,70 @@
        that:
            - "azure_managed_disk | length > 0"
 
+ - name: Create virtual network
+   azure_rm_virtualnetwork:
+       resource_group: "{{ resource_group }}"
+       name: testvm001
+       address_prefixes: "10.10.0.0/16"
+
+ - name: Add subnet
+   azure_rm_subnet:
+       resource_group: "{{ resource_group }}"
+       name: testvm001
+       address_prefix: "10.10.0.0/24"
+       virtual_network: testvm001
+
+ - name: Create public ip
+   azure_rm_publicipaddress:
+       resource_group: "{{ resource_group }}"
+       allocation_method: Static
+       name: testvm001
+
+ - name: Create security group
+   azure_rm_securitygroup:
+       resource_group: "{{ resource_group }}"
+       name: testvm001
+
+ - name: Create NIC
+   azure_rm_networkinterface:
+       resource_group: "{{ resource_group }}"
+       name: testvm001
+       virtual_network: testvm001
+       subnet: testvm001
+       public_ip_name: testvm001
+       security_group: testvm001
+
+ - name: Create virtual machine
+   azure_rm_virtualmachine:
+       resource_group: "{{ resource_group }}"
+       name: testvm001
+       admin_username: adminuser
+       admin_password: Password123!
+       os_type: Linux
+       managed_disk_type: Premium_LRS
+       vm_size: Standard_DS1_v2
+       network_interfaces: testvm001
+       image:
+         offer: UbuntuServer
+         publisher: Canonical
+         sku: 16.04-LTS
+         version: latest
+ 
+ - name: Mount the disk to virtual machine
```

Comment:
It is best to run 3 tests for an operation with asserts after each operation;

* check mode
* normal
* idempotent

For example I would do

```
- name: Mount the disk to virtual machine (check)
  azure_rm_managed_disk:
    ...
  register: mounted_check
  check_mode: yes

- name: get details of mounted disk (check mode)
  azure_rm_managed_disk_facts:
    ...
  register: mounted_result_check

- name: assert result of mount the disk to virtual machine
  assert:
    that:
    - mounted_check|changed
    - mounted_result_check.managed_by != "..."
    ....
```

Repeat these 3 steps 2 more times, 1 not in check mode and then another to prove that Ansible does not say a change is available.

Reply:
Are you able to create a check mode and idempotent test task for mounting the disk to the virtual machine